core/src/main/scala/kafka/zookeeper/ZooKeeperClient.scala

```diff
@@ -353,6 +353,17 @@ class ZooKeeperClient(connectString: String,
     info("Closed.")
   }
 
+  /**
+   * Wait up to the provided timeoutMs this ZooKeeperClient to close.
+   * Supplying a timeoutMs of 0 means to wait forever.
+   */
```

Comment:
We should explain the difference between this and `close` (including a javadoc for `close`). Also, I wonder if we should use a different name for this one (instead of just an overload or default parameter for the timeout).

@junrao Any thoughts on what the default behavior of `close` should be? It currently closes the connection and returns. But if we pass a timeout, then it tries to join the send and event thread. We should probably do the same with our scheduler thread, in that case.